server/handlers/project_command_output_handler.go

```diff
@@ -130,6 +137,12 @@ func (p *AsyncProjectCommandOutputHandler) Register(jobID string, receiver chan
 
 func (p *AsyncProjectCommandOutputHandler) Handle() {
 	for msg := range p.projectCmdOutput {
+		if msg.OperationComplete {
+			p.updateOutputBufferOperationStatus(msg.JobID)
+			p.closeChannelsForJob(msg.JobID)
```

Comment:
Honestly you can combine these two into 1 method and call it `completeJob` or something like that.

Reply:
Yup that makes sense! I've combined them both into one method `completeJob()`